### PR TITLE
OOP Library: Project setup

### PR DESCRIPTION
**Pull request by kkolade**
# In this PR

- I create a class Person with the following:
   - Instance vars: @id, @name, and @age.
   - Constructor with name, age, and parent_permission as parameter
   - Getters for @id, @name, and @age.
   - Setters for @name and @age.
   - Private method of_age? that returns true if @age is greater or equal to 18 and false otherwise.
   - Public method can_use_services? that returns true if person is of age or if they have permission from parents.
- I create a class Student with the following:
   - That inherits from Person.
   - has a constructor that extends Person's constructor by adding @classroom and a parameter for it.
   - A method play_hooky that returns "¯\(ツ)/¯".
 - I create a class Teacher with the following:
   - Inherits from Person.
   -  has a constructor extends Persons's constructor by adding @specialization and a parameter for it.
   - Overrides can_use_services? so it always returns true.
- Each class should be saved in a separate file.

### CODE REVIEW

**Review by Mihndim2020 — COMMENTED** [review]
# STATUS: INVALID!!!  :x: :x: :x: 

Hi @kkolade,

While you made a great effort in this project, unfortunately, I cannot proceed to review your code, please make sure your implement the following changes before you resubmit for review. 

## Required Changes ♻️
_Check the comments below._
- [ ] Since you have not hosted the project online, it will be  nice to remove the `Live demo` section in your readme file completely as the will make your readme file look more professional. 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear or you can get to me on slack  using this handle @mihndim.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

_Invalid Code Review Request **does not count** into the code reviews limit._

File: student.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+require './person'
+
+class Student < Person
+  def initialize(classroom)
```
Comment (GzimAsani) [inline student.rb]: - [ ] The student should also initialize all of the parameters as the person 
``` ruby
 def initialize(age, classroom, name = "Unknown", parent_permission = true)  
```

File: teacher.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+require './person'
+
+class Teacher < Person
+  def initialize(specialization)
+    super()
+    @specialization = specialization
+  end
```
Comment (GzimAsani) [inline teacher.rb]: - [ ] You need to include all the previous inputs from the person together with specialization, same thing with super()
``` ruby
  def initialize(age, specialization, name = "Unknown", parent_permission = true)
    super(age, name, parent_permission)
    @specialization = specialization
  end

```

**Review by GzimAsani — CHANGES_REQUESTED** [review]
### Hello, @kkolade                   :wave:   
 Good job so far,  done There are some issues that you still need to work on to go to the next project but you are almost there!
 
 ## Highlights
 * Good job with your person class :+1: 

 ## Required Changes 

- [ ] Check the comments below :arrow_double_down: 

 ## Optional suggestions
 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding! clapclapclap
 
 Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.
 
 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
+class Person
+  def initialize(age, name = 'Unknown')
```
Comment (Kamba56) [inline person.rb:2]: - [Optional] I noticed you are passing the parents_permission as a parameter in the super method but the Person class doesn't accept that parameter. Kindly add the `parent_permission` as an argument to the Person constructor and set it to true.

**Review by Kamba56 — APPROVED** [review]
Hi @kkolade, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏
## Highlights
- [x] Great job inheriting from Person class!👏👏👏
- [x] Great job using attr_accessor for setter and getter for name and age!👏👏👏
- [x] Great job creating classes in ruby!👏👏👏

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**